### PR TITLE
Add sqllogictest for modeling-rs end-to-end test

### PR DESCRIPTION
**Pull request by goldmedal**
# Description
Basically, I copied [the sqllogictest in DataFusion](https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest) as our testing framework. Just remove the postgres test and the test cases.
The e2e test uses DataFusion as the query runner. So I also registered the MDL dataset before running tests. 

You can run this e2e test by the command
```
cargo test --test sqllogictests
```

# Follow-up issues
- Add readme for this module
- Handle the qualified table name https://github.com/Canner/wren-engine/pull/604#discussion_r1630922098

### CODE REVIEW

File: wren-modeling-rs/sqllogictest/src/test_context.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        self.test_dir.as_ref().expect("testdir not enabled").path()
+    }
+
+    /// Returns a reference to the internal SessionContext
+    pub fn session_ctx(&self) -> &SessionContext {
+        &self.ctx
+    }
+}
+
+pub async fn register_ecommerce_table(test_ctx: &TestContext) {
```
Comment (goldmedal) [inline wren-modeling-rs/sqllogictest/src/test_context.rs:109]: I created and registered the ecommerce case here.

File: wren-modeling-rs/sqllogictest/src/test_context.rs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                .column(ColumnBuilder::new("order_id", "varchar").build())
+                .column(ColumnBuilder::new("purchase_timestamp", "timestamp").build())
+                .build(),
+        )
+        .build();
+    let analyzed_mdl = Arc::new(AnalyzedWrenMDL::analyze(manifest));
+    ctx.state().with_analyzer_rules(vec![
+        Arc::new(ModelAnalyzeRule::new(Arc::clone(&analyzed_mdl))),
+        Arc::new(ModelGenerationRule::new(Arc::clone(&analyzed_mdl))),
+    ]);
```
Comment (goldmedal) [inline wren-modeling-rs/sqllogictest/src/test_context.rs:162]: I registered the wren analysis rule here.

File: wren-modeling-rs/sqllogictest/test_sql_files/model.slt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+statement ok
+SELECT * from orders
```
Comment (goldmedal) [inline wren-modeling-rs/sqllogictest/test_sql_files/model.slt:2]: This test should be `SELECT * from wrenai.default.orders` actually.
There is an issue about analyzing a qualified table name. I'll address this in next PR.

File: wren-modeling-rs/core/src/mdl/builder.rs [hunk 4]
```diff
@@ -10,6 +10,12 @@ pub struct ManifestBuilder {
...
 
+impl Default for ManifestBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl ManifestBuilder {
     pub fn new() -> Self {
         Self {
```
Comment (grieve54706) [inline wren-modeling-rs/core/src/mdl/builder.rs:21]: Why add a new impl `Default` for `default()`? Why not just make the `default()` into the impl `ManifestBuilder`?

Comment (goldmedal) [reply wren-modeling-rs/core/src/mdl/builder.rs:21]: It's a standard trait in Rust. It's convenience for some design pattern.
https://rust-unofficial.github.io/patterns/idioms/default.html

Comment (grieve54706) [reply wren-modeling-rs/core/src/mdl/builder.rs:21]: Thanks for your information.

**Review by grieve54706 — APPROVED** [review]
LGTM

**Pull request merged**